projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/components/RunJob.tsx

```diff
@@ -71,41 +71,139 @@ export default class RunJobDialog extends Component<IJobPathProp> {
-      ></RunJobDialog>
-    ),
+    body: <RunJobDialog jobPath={jobData.get(VdkOption.PATH)!}></RunJobDialog>,
     buttons: [Dialog.okButton(), Dialog.cancelButton()]
   });
   if (result.button.accept) {
     let { message, status } = await jobRunRequest();
-        if (status) {
-          showDialog({
-            title: 'Run Job',
-            body: <div className='vdk-run-dialog-message-container'>
-            <p className='vdk-run-dialog-message'>The job was executed successfully!</p>
-          </div>,
-            buttons: [Dialog.okButton()]
-          });
-        }
-        else{
-          message = "ERROR : " + message;
-          const  errorMessage = new VdkErrorMessage(message);
-          showDialog({
-            title: 'Run Job',
-            body: <div  className="vdk-run-error-message ">
+    if (status) {
+      showDialog({
+        title: 'Run Job',
+        body: (
+          <div className="vdk-run-dialog-message-container">
+            <p className="vdk-run-dialog-message">
+              The job was executed successfully!
+            </p>
+          </div>
+        ),
+        buttons: [Dialog.okButton()]
+      });
+    } else {
+      message = 'ERROR : ' + message;
+      const errorMessage = new VdkErrorMessage(message);
+      if (
+        !docManager ||
+        !(await handleErrorsProducedByNotebookCell(errorMessage, docManager))
+      ) {
+        showDialog({
+          title: 'Run Job',
+          body: (
+            <div className="vdk-run-error-message ">
               <p>{errorMessage.exception_message}</p>
               <p>{errorMessage.what_happened}</p>
               <p>{errorMessage.why_it_happened}</p>
               <p>{errorMessage.consequences}</p>
               <p>{errorMessage.countermeasures}</p>
-            </div>,
-            buttons: [Dialog.okButton()]
-          });
-        }
+            </div>
+          ),
+          buttons: [Dialog.okButton()]
+        });
+      }
+    }
   }
 }
+
+export const findFailingCellId = (message: String): string => {
+  const regex = /cell_id:([0-9a-fA-F-]+)/;
+  const match = message.match(regex);
+  if (match) return match[1];
+  return '';
+};
+
+/**
+ * Seperate handling for notebook errors - option for the user to navigate to the failing cell when error is produced
+ */
+export const handleErrorsProducedByNotebookCell = async (
+  message: VdkErrorMessage,
+  docManager: IDocumentManager
+): Promise<boolean> => {
+  const failingCellId = findFailingCellId(message.what_happened);
+  if (failingCellId) {
+    const { path: nbPath, cellIndex: failingCellIndex } = await getNotebookInfo(
+      failingCellId
+    );
+    if (nbPath) {
+      const navigateToFailingCell = async () => {
+        const notebook = docManager.openOrReveal(nbPath);
+        if (notebook) {
+          await notebook.revealed; // wait until the DOM elements are fully loaded
+          const children = Array.from(notebook.node.children!);
+          if (children) {
```

Comment:
You don't need this check. Correct me if I'm wrong, but you'll always get an array from the above line, even if it's an empty one. Then you'll run the forEach on an empty array, which will do nothing.